debug/log/kubernetes/log.go

```diff
@@ -0,0 +1,34 @@
+package kubernetes
+
+import "github.com/micro/go-micro/debug/log"
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+func write(l log.Record) error {
+	m, err := json.Marshal(l)
+	if err == nil {
+		_, err := fmt.Fprintf(os.Stderr, "%s", m)
+		return err
+	}
+	return err
+}
+
+type klogStreamer struct {
+	streamChan chan log.Record
+}
+
+func (k *klogStreamer) Chan() <-chan log.Record {
+	if k.streamChan == nil {
+		k.streamChan = make(chan log.Record)
+	}
+	return k.streamChan
+}
+
+func (k *klogStreamer) Stop() error {
+	close(k.streamChan)
```

Comment:
Stop() may be called multiple times so you want to do this sort of select statement. But also because something might be sending to streamChan it may panic if you close it so you want to close a stop channel instead.

```
select {
case <-k.stop:
  return nil
default:
  close(k.stop)
```

Reply:
Yep, I'll worry about that while implementing